src/julia.h

```diff
@@ -397,6 +397,7 @@ typedef struct {
     unsigned exportp:1;
     unsigned imported:1;
     unsigned deprecated:1;
+    jl_sym_t *deprecation_preferred_name;  // non-NULL when new type incompatible with old
```

Comment:
Note that currently the `jl_binding_t` is 5 pointer size each of them at least 4bytes aligned so we can just use the pointer tag to store the 3 flags. (We can have up to 10 flags this way.) This is possible since a `jl_binding_t` is not a first class object. Might worth checking memory usage etc to decide if this is worth doing.